server.js

```diff
@@ -0,0 +1,221 @@
+//require express 
+var express = require("express");
+//require body parser
+var bodyParser = require("body-parser");
+
+var app = express();
+
+//require mongoose
+var mongoose = require("mongoose");
+
+//connect the data base
+mongoose.connect("mongodb://localhost/blog-posts-app");
+
+//require schemas
+var Post = require("./models/blogpost");
+var Comment = require("./models/comment");
+var User = require('./models/user');
+
+//apply body parser package
+app.use(bodyParser.urlencoded({extended:true}));
+
+//connect public folder
+app.use(express.static(__dirname + "/public"));
+
+//log in settings
+var cookieParser = require('cookie-parser');
```

Comment:
FYI it's best to put all of your 'require' statements at the top, so they're all together and set up before you do any other code in the server.